internal/cli/install.go

```diff
@@ -520,21 +522,26 @@ func installRunner(
 	if odc, ok := p.(serverinstall.OnDemandRunnerConfigProvider); ok {
 		s = sg.Add("Registering on-demand runner configuration...")
 
-		odr := odc.OnDemandRunnerConfig()
-		if odr != nil {
-			_, err = client.UpsertOnDemandRunnerConfig(ctx, &pb.UpsertOnDemandRunnerConfigRequest{
-				Config: odr,
-			})
+		if odrConfig == nil {
+			odrConfig = odc.OnDemandRunnerConfig()
+		}
 
-			if err != nil {
-				s.Update("Error creating ondemand runner: %s", err)
-				s.Status(terminal.StatusError)
-			} else {
-				s.Update("Registered ondemand runner!")
-				s.Status(terminal.StatusOK)
-			}
+		version, err := client.GetVersionInfo(ctx, &empty.Empty{})
+		// The expected naming convention is "<platform>-default-<waypoint_version>
+		odrConfig.Name = odrConfig.Name + "-default-" + version.Info.Version
+		if err != nil {
+			ui.Output("Error getting version: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
+		}
+
+		_, err = client.UpsertOnDemandRunnerConfig(ctx, &pb.UpsertOnDemandRunnerConfigRequest{
+			Config: odrConfig,
+		})
+		if err != nil {
+			s.Update("Error creating ondemand runner: %s", err)
+			s.Status(terminal.StatusError)
```

Comment:
Same feedback here, return err? I recommend making this a `ui.Output` instead of a step group too